tests/ha/single_pod_restart/test_server_pod_restart_api.py

```diff
@@ -1290,5 +1419,200 @@ def test_obj_ver_suspension_server_pod_restart(self):
+        args = {'chk_null_version': True, 'is_unversioned': True, 'file_path':
+            self.multipart_obj_path, 'enable_ver': True, 's3_ver': self.s3_ver}
+        resp = self.ha_obj.crt_bkt_put_obj_enbl_ver(event, self.s3_test_obj, self.bucket_name,
+                                                 self.object_name, **args)
+        assert_utils.assert_true(resp[0], resp[1])
+        LOGGER.info("Step 1: Created bucket and uploaded object %s of %s size. Enabled "
+                    "versioning on %s.", self.object_name, self.f_size, self.bucket_name)
+        self.version_etag.update({self.bucket_name: []})
+        self.version_etag[self.bucket_name].extend(resp[1])
+        self.version_etag.update({"obj_name": self.object_name})
+        self.version_etag.update({"s3_ver": self.s3_ver})
+        self.is_ver = True
+        bucket_list = list()
+        bucket_list.append(self.bucket_name)
+        LOGGER.info("Step 2: Upload same object %s after enabling versioning and list & verify "
+                    "the same.", self.object_name)
+        args = {'file_path': self.multipart_obj_path}
+        resp = self.ha_obj.parallel_put_object(event, self.s3_test_obj, self.bucket_name,
+                                               self.object_name, **args)
+        assert_utils.assert_true(resp[0], f"Upload Object failed {resp[1]}")
+        self.version_etag[self.bucket_name].extend(resp[1])
+        resp = self.ha_obj.list_verify_version(self.s3_ver, self.bucket_name, self.version_etag[
+            self.bucket_name])
+        assert_utils.assert_true(resp[0], resp[1])
+        LOGGER.info("Step 2: Uploaded same object %s after enabling versioning and listed & "
+                    "verified the same.", self.object_name)
+
+        LOGGER.info("Step 3: Suspend versioning on %s.", self.bucket_name)
+        resp = self.s3_ver.put_bucket_versioning(bucket_name=self.bucket_name, status="Suspended")
+        assert_utils.assert_true(resp[0], resp)
+        LOGGER.info("Step 3: Suspended versioning on %s.", self.bucket_name)
+
+        LOGGER.info("Step 4: Get versions of %s with & without VersionID & verify etags.",
+                    self.object_name)
+        resp = self.ha_obj.parallel_get_object(event=event, s3_ver_obj=self.s3_ver,
+                                               bkt_name=self.bucket_name, obj_name=self.object_name,
+                                               ver_etag=self.version_etag[self.bucket_name])
+        assert_utils.assert_true(resp[0], f"Get Object with versionID failed {resp[1]}")
+        resp = self.s3_test_obj.get_object(self.bucket_name, self.object_name)
+        latest_vetag = self.version_etag[self.bucket_name][-1]
+        latest_v = list(latest_vetag.keys())[0]
+        etag = list(latest_vetag.values())[0]
+        if resp[1]["VersionId"] != latest_v:
+            assert_utils.assert_true(False, "Get Object without VersionID does not match with "
```

Comment:
instead of without version ID, add with version id null. Check add all the places.

Reply:
Why NULL, last updated can be anything. This step is a get operation with and without specifying VersionID, which will fetch the last version ID if we want to fetch without Version ID.

Reply:
Then mention without passing/specifying version ID.

Reply:
Already mentioned in step statement.

Reply:
But the error message gives different meaning. Get object without version ID is implying that object doesn't have any version ID.

Reply:
Updated